block-storage/src/main/scala/coop/rchain/blockstorage/deploy/DeployStorage.scala

```diff
@@ -0,0 +1,14 @@
+package coop.rchain.blockstorage.deploy
+
+import coop.rchain.casper.protocol.DeployData
+import coop.rchain.crypto.signatures.Signed
+
+trait DeployStorage[F[_]] {
+  def add(deploys: List[Signed[DeployData]]): F[Unit]
+  def remove(deploys: List[Signed[DeployData]]): F[Int]
```

Comment:
I don't think there is much to report apart from some errors I would consider fatal (e.g. our scodec codec can't handle some types of deploys as in your comment below).

As for returning `int` in `remove` signature, you are right in that it does not tell us much. I think it is okay as the only it is used in is logging so that you could see whether block finalization removed any old deploys in the logs.